osquery/tables/system/windows/chassis_info.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genChassisInfo(QueryContext& context) {
+  QueryData results;
+
+  WmiRequest wmiSystemReq("select * from Win32_SystemEnclosure");
+  const auto& wmiResults = wmiSystemReq.results();
+
+  //check if the results are empty and return a warning if so
+  if (wmiResults.empty()) {
+    LOG(WARNING) << wmiSystemReq.getStatus().getMessage();
+  }
+
+  for (const auto& data : wmiResults) {
+    Row r;
+    bool boolean = false;
```

Comment:
The way the variable gets reused seems weird to me. But clearly there's prior art for it. 

I like you're suggestion in the other thread to rename to `isPresent`

Reply:
I'm not a c++ wizard as much as I want to be, but the challenge here is taking a bool from wmiResults and translating it into a String for the table. Never tried - but maybe I can cast bool as a string? I'll look into that.

Reply:
Reminder for myself: don't forget to rebase this branch cause its 2 years outdayed.